File: oocpp/code/software_sales.cpp
/*
	Carlos Vigil
	Dr. T. Laurasi
	OOC++
	March 20th, 2017

	Chapter 4 Code Assignment
	Software Sales
*/

#include <iostream>
#include <string>
using namespace std;

int main () {
	// variable declaration
	int individual_package_price = 99,
		user_input_quantity,
		total_retail_price,
		total_discount,
		total_discounted_price;
	string user_prompt = "How many packages are you purchasing?",
		result_preface = "The discounted price: $";

	// Ask the user to input a quantity
	cout << user_prompt << endl;
	cin >> user_input_quantity;

	// wonder what it means
	if (user_input_quantity < 0) {
		cout << "Nope, can't buy those back." << endl;
		main(); // run again
	} else if (user_input_quantity > 99) {
		cout << "That gets you a 50% discount!" << endl;
		// do the math
		total_retail_price = individual_package_price * user_input_quantity;
		total_discount = 0.50 * total_retail_price;
		total_discounted_price = total_retail_price - total_discount;
		// output discounted price
		cout << result_preface << total_discounted_price << endl;
	} else if (user_input_quantity > 49) {
		cout << "You get 40% off for that many." << endl;
		total_retail_price = individual_package_price * user_input_quantity;
		total_discount = 0.40 * total_retail_price;
		total_discounted_price = total_retail_price - total_discount;
		cout << result_preface << total_discounted_price << endl;
	} else if (user_input_quantity > 19) {
		cout << "30% off regular price is your discount." << endl;
		total_retail_price = individual_package_price * user_input_quantity;
		total_discount = 0.30 * total_retail_price;
		total_discounted_price = total_retail_price - total_discount;
		cout << result_preface << total_discounted_price << endl;
	} else if (user_input_quantity > 9) {
		cout << "20% off for the extra mini enterprise pack." << endl;
		total_retail_price = individual_package_price * user_input_quantity;
		total_discount = 0.20 * total_retail_price;
		total_discounted_price = total_retail_price - total_discount;
		cout << result_preface << total_discounted_price << endl;
	} else {
		cout << "No discount for that many, but you get stickers." << endl;
		total_retail_price = individual_package_price * user_input_quantity;
		total_discount = 0.50 * total_retail_price;
		total_discounted_price = total_retail_price - total_discount;
		cout << result_preface << total_discounted_price << endl;
	}
}

File: java/code/SoftwareSales.java
import java.util.Scanner;
/*
    Carlos Vigil
    Dr. Horvath
    Java
    February 7th, 2017
*/

// Chapter 3, #8 Software Sales
public class SoftwareSales
{
    public static void main(String[] args)
    {
        int bargain; // the user entered number of packages
        String prompt = "How many packages are you purchasing?\n"; // ask user
        Scanner keyboard = new Scanner(System.in); // reads input

        // Ask the user for a number
        System.out.print(prompt);
        bargain = keyboard.nextInt();

        // wonder what it means
        if (bargain > 99) {
            System.out.println("That gets you a 50% discount!");
        } else if (bargain > 49) {
            System.out.println("You get 40% off for that many.");
        } else if (bargain > 19) {
            System.out.println("30% off regular price is your discount.");
        } else if (bargain > 9) {
           System.out.println("20% off for the extra mini enterprise pack."); 
        } else {
            System.out.println("No discount for that many, but you get stickers.");
        }
    }
}


/*
    8. Software Sales
    A software company sells a package that retails for $99. Quantity discounts
    are given according to the following table:
    Quantity Discount
        10–19 20%
        20–49 30%
        50–99 40%
        100 or more 50%
    Write a program that asks the user to enter the number of packages
    purchased. The program should then display the amount of the discount (if
    any) and the total amount of the purchase after the discount.
*/